ntoskrnl/mm/balance.c

```diff
@@ -404,7 +406,14 @@ MiBalancerThread(PVOID Unused)
             while (InitialTarget != 0);
 
             if (Status == STATUS_WAIT_0)
-                InterlockedDecrement(&PageOutThreadActive);
+            {
+#if DBG
+                LONG Active = InterlockedExchange(&PageOutThreadActive, 0);
+                ASSERT(Active == 1);
```

Comment:
Is it possible to do it this compact?
```suggestion
                ASSERT(InterlockedExchange(&PageOutThreadActive, 0) == 1);
```

Reply:
Not sure. I haven't seen something like this used anywhere.